content/knowledge/consensus_algorithms.md

```diff
@@ -0,0 +1,285 @@
+This is the transitive closure property that these types of systems typically use to fight against Sybil attacks.
+
+A living example of such a network is the Internet itself.
+It consists of independent nodes that know about a few other nodes local to them, but the sets overlap with each other and every node is reachable through any other by routes and gateways.
+In the same manner, an FBA network resembles inter-domain network routing with no central authority dictating or arbitraging agreements.
+Quorum slices can be perceived as analogous to internet networks in terms of reachability.
+Servers in the internet networks can be reached via requests from neighboring networks, using gateways and hubs, making each server accessible through any client.
+In quorums, nodes are reachable through their neighboring nodes, which might be in another quorum, but because of the nodes' connections between each other, each node can be reached theoretically from any other.
+The Internet's nearly complete transitive reachability suggests it can likewise ensure a worldwide consensus with FBA.
+
+Voting in Stellar consists of two phases: nomination and balloting.
+
+At the beginning of the nomination phase, a node randomly votes for a statement on a ledger version it receives from another node in the network.
+The nodes cast out their nomination votes via messages to their peers, as well as the nominations of their peers, meaning separate federated votes are batched together.
+Voting against a value is not defined and a node can nominate as many statements as it likes so after the FBA process is run on each node, a nominee is confirmed and it becomes a _candidate_.
+
+The nomination may produce multiple confirmable candidates, so SCP requires the application layer to supply some method for combining them into a single composite, with the only requirement for this method to be deterministic.
+
+SCP is an asynchronous protocol and nodes are not coordinated by time, but by the messages they exchange.
+Thus, from a node’s perspective, it is not clear when the nomination process has ended.
+During the nomination, it cannot be said which of the composites is the final one, but this is expected because the purpose is to produce several virtuous candidates for the balloting.
+
+In the balloting phase nodes exchange their ballots on created composites and repeatedly conduct federated votes on statements about ballots.
+From the point of view of a given node, a consensus is reached when it finds a ballot for which it can confirm (find a quorum accepting) a statement, and then it can be sure that every other well-behaved node has committed, or inevitably will commit to, the same value.
+
+Nodes can be categorized as either well-behaved or ill-behaved. // not sure if we need this part here or in the glossary maybe or in another named section
```

Comment:
Was this forgotten or is it a plan to take care of it later on?
(I'm fine either way, just wondering if it slipped through the cracks :))

Reply:
Well this is a text block that I left here but I am not 100% convinced it should be here - specifically the explanation about well-behaved and ill-behaved nodes - not sure if its too much details.

Reply:
Decided to leave it in the text; IMO it will bring good context